app/models/api_key.rb

```diff
@@ -2,15 +2,18 @@ class ApiKey < ApplicationRecord
   API_SCOPES = %i[index_rubygems push_rubygem yank_rubygem add_owner remove_owner access_webhooks show_dashboard].freeze
   APPLICABLE_GEM_API_SCOPES = %i[push_rubygem yank_rubygem add_owner remove_owner].freeze
 
-  belongs_to :user
+  belongs_to :user, inverse_of: :api_keys
+  belongs_to :owner, polymorphic: true
 
   has_one :api_key_rubygem_scope, dependent: :destroy
   has_one :ownership, through: :api_key_rubygem_scope
   has_one :oidc_id_token, class_name: "OIDC::IdToken", dependent: :restrict_with_error
   has_one :oidc_api_key_role, through: :oidc_id_token, inverse_of: :api_key
   has_many :pushed_versions, class_name: "Version", inverse_of: :pusher_api_key, foreign_key: :pusher_api_key_id, dependent: :nullify
 
-  validates :user, :name, :hashed_key, presence: true
+  before_validation :set_owner_from_user
+
+  validates :user, :owner, :name, :hashed_key, presence: true
```

Comment:
I think both user and owner are duplicates of the default rails behavior.

https://github.com/rails/rails/blob/d39db5d1891f7509cde2efc425c9d69bbb77e670/activerecord/lib/active_record/associations/builder/belongs_to.rb#L127

Reply:
iirc belongs_to has presence validation by default, but has_one does not

Reply:
Yes, I'm suggesting:
```suggestion
  validates :name, :hashed_key, presence: true
```